tests/solvers/test_solvers.py

```diff
@@ -33,6 +43,17 @@ def test_scipyopt(self, rb):
         solver.fit(data=None, options={'maxiter': 20})
         assert np.abs(solver.fun_val_opt) < 1e-5
 
+    @pytest.mark.parametrize('curve_fun', [ln_gaussian_pdf, ln_gaussian_cdf, gaussian_pdf, gaussian_cdf])
+    def test_scipyopt_core_mdoel(self, curve_fun):
+        params_set, params_true, _ = simulate_params(1)
+        data = simulate_data(curve_fun, params_true)
+        core_model = CoreModel(params_set, curve_fun, normal_loss)
+        solver = ScipyOpt(core_model)
+        solver.fit(data=data, options={'maxiter': 50})
+        y_pred = solver.predict(t=data[0]['t'].to_numpy())
+        y_true = data[0]['obs'].to_numpy()
+        assert np.linalg.norm(y_pred - y_true) / np.linalg.norm(y_true) < 1e-2
+        
```

Comment:
Can we also do one that tests the predict for multiple groups?

Reply:
Yes I can do that, one thing I was thinking is how we would check the solution, since we do not have random effects summing to zero constraint...

Reply:
Oh interesting.. Hmmm. Thoughts @zhengp0 ?